sql/2022/markup/document_trends.sql

```diff
@@ -0,0 +1,46 @@
+#standardSQL
+# summary_pages trends grouped by device
+
+WITH summary_requests AS (
+  SELECT
+    '2019' AS year,
+    _TABLE_SUFFIX AS client,
+    *
+  FROM
+    `httparchive.summary_pages.2019_07_01_*`
+  UNION ALL
+  SELECT
+    '2020' AS year,
+    _TABLE_SUFFIX AS client,
+    *
+  FROM
+    `httparchive.summary_pages.2020_08_01_*`
+  UNION ALL
+  SELECT
+    '2021' AS year,
+    _TABLE_SUFFIX AS client,
+    *
+  FROM
+    `httparchive.summary_pages.2021_07_01_*`
+  UNION ALL
+  SELECT
+    '2022' AS year,
+    _TABLE_SUFFIX AS client,
+    * EXCEPT (metadata)
+  FROM
+    `httparchive.summary_pages.2022_06_01_*`
+)
```

Comment:
This could be shortened to this:

```sql
  SELECT
    SUBSTR(_TABLE_SUFFIX, 0, 4) AS year,
    SUBSTR(_TABLE_SUFFIX, 12) AS client,
    * EXCEPT (metadata)
  FROM
    `httparchive.summary_pages.*`
  WHERE
    SUBSTR(_TABLE_SUFFIX, 0, 10) in ('2019_07_01', '2020_08_01', '2021_07_01', '2022_06_01')
```

Reply:
This is a rerun+addition of a previous year's query so not looking to refactor it.

Reply:
I don’t think it’s a real “refactor”. It’s just a simplification of the queries to stop this getting out of hand as more and more years are added. But not that big a deal.